plugins/modules/digital_ocean_kubernetes.py

```diff
@@ -0,0 +1,421 @@
+    type: bool
+    required: no
+    default: False
+  surge_upgrade:
+    description:
+      - A boolean value indicating whether surge upgrade is enabled/disabled for the cluster.
+      - Surge upgrade makes cluster upgrades fast and reliable by bringing up new nodes before destroying the outdated nodes.
+    type: bool
+    required: no
+    default: False
+  tags:
+    description:
+      - A flat array of tag names as strings to be applied to the Kubernetes cluster.
+      - All clusters will be automatically tagged "k8s" and "k8s:$K8S_CLUSTER_ID" in addition to any tags provided by the user.
+    type: list
+    required: no
+  maintenance_policy:
+    description:
+      - An object specifying the maintenance window policy for the Kubernetes cluster (see table below).
+    type: dict
+    required: no
+  node_pools:
+    description:
+      - An object specifying the details of the worker nodes available to the Kubernetes cluster (see table below).
+    type: list
+    required: yes
+    default:
+      - name: worker-pool
+        size: s-1vcpu-2gb
+        count: 1
+        tags: []
+        labels: {}
+  vpc_uuid:
+    description:
+      - A string specifying the UUID of the VPC to which the Kubernetes cluster will be assigned.
+      - If excluded, the cluster will be assigned to your account's default VPC for the region.
+    type: str
+    required: no
+  return_kubeconfig:
+    description:
+      - Controls whether or not to return the C(kubeconfig).
+    type: bool
+    required: no
+    default: False
+  wait:
+    description:
+      - Wait for the cluster to be running before returning.
+    type: bool
+    required: no
+    default: True
+  wait_timeout:
+    description:
+      - How long before wait gives up, in seconds, when creating a cluster.
+    type: int
+    default: 600
+requirements:
+    - python >= 2.6
+'''
+
+
+EXAMPLES = r'''
+- name: Create a new DigitalOcean Kubernetes cluster in New York 1
+  community.digitalocean.digital_ocean_kubernetes:
+    state: present
+    oauth_token: "{{ lookup('env', 'DO_API_TOKEN') }}"
+    name: hacktoberfest
+    region: nyc1
+    node_pools:
+      - name: hacktoberfest-workers
+        size: s-1vcpu-2gb
+        count: 3
+    return_kubeconfig: yes
+    wait_timeout: 600
+    register: my_cluster
+
+- name: Show the kubeconfig for the cluster we just created
+  debug:
+    msg: "{{ my_cluster.data.kubeconfig }}"
+
+- name: Destroy (delete) an existing DigitalOcean Kubernetes cluster
+  community.digitalocean.digital_ocean_kubernetes:
+    state: absent
+    oauth_token: "{{ lookup('env', 'DO_API_TOKEN') }}"
+    name: hacktoberfest
+'''
+
+
+# Digital Ocean API info https://developers.digitalocean.com/documentation/v2/#kubernetes
+# The only variance from the documented response is that the kubeconfig is (if return_kubeconfig is True) merged in at data['kubeconfig']
```

Comment:
I'm wondering if we really want to return the kubeconfig, or if it's better practice to get the kubeconfig in a separate explicit task? Genuinely wondering, because I know with EKS in AWS, I generally have a separate task after the cluster is created to generate/retrieve the kubeconfig using the AWS CLI (since there isn't a direct Ansible module for it).

Reply:
```            return_kubeconfig=dict(type='bool', default=False),```

Was my thinking ... I'm fine either way though. Few more passes through the docstring battle and I'm going to 💤 .